plugins/elasticsearch/encoders.go

```diff
@@ -326,24 +350,52 @@ func (e *ESJsonEncoder) Encode(pack *PipelinePack) (output []byte, err error) {
 		case "hostname":
 			writeStringField(first, &buf, e.fieldMappings.Hostname, m.GetHostname())
 		case "fields":
-			for _, field := range m.Fields {
-				raw := false
-				if len(e.rawBytesFields) > 0 {
-					for _, raw_field_name := range e.rawBytesFields {
-						if *field.Name == raw_field_name {
-							raw = true
+			if len(e.dynamicFields) > 0 {
+				fmt.Printf("has dynamic fields %s", e.dynamicFields)
+				for _, field := range m.Fields {
```

Comment:
Ignore me. You fetch fields by name in go code by using [message.FindAllFields](https://github.com/mozilla-services/heka/blob/dev/message/message.go#L501), but I looked at the implementation of that method and realized that it's just iterating through the fields every time you call it, so it doesn't help after all. :P
Given that, though, I wonder if it doesn't make sense to merge both of the loops (i.e. len(`e.dynamicFields > 0` and not) into one. It means an extra 'if' check on each iteration of the loop when there are no dynamic fields, but that's not a lot of cycles, and you can pre-compute `e.dynamicFields > 0` before the loop so you're just checking a boolean value.